tests/benchmark/test_bloatnet.py

```diff
@@ -0,0 +1,618 @@
+            deploy_bytecode += Op.JUMPDEST
+
+        assert len(deploy_bytecode) == max_contract_size, (
+            f"Contract size mismatch: {len(deploy_bytecode)}"
+        )
+
+        # Calculate the CREATE2 address (using base init_code for consistent addresses)
+        create2_addr = compute_create2_address(
+            address=factory_address, salt=salt, initcode=bytes(init_code)
+        )
+        # Deploy at the calculated address with unique bytecode
+        pre[create2_addr] = Account(code=deploy_bytecode)
+        deployed_contracts.append(create2_addr)
+
+    # Create the attack contract that calculates CREATE2 addresses and calls operations
+    attack_code = Bytecode()
+
+    # Pre-compute the CREATE2 calculation bytecode
+    create2_calc = generate_create2_address_calculation(factory_address, init_code_hash)
+
+    # Main loop: iterate through all contracts
+    attack_code += Op.PUSH0  # Counter starts at 0
+    attack_code += While(
+        body=(
+            # Calculate CREATE2 address for current salt (counter)
+            Op.DUP1  # Duplicate counter (salt)
+            + create2_calc  # Calculate CREATE2 address (consumes salt, leaves address)
+            # Call BALANCE (cold access) - this warms the account
+            + Op.DUP1  # Duplicate address
+            + Op.BALANCE  # Get balance
+            + Op.POP  # Discard balance result
+            # Call EXTCODESIZE (warm access) - hoping cache was evicted
+            + Op.EXTCODESIZE  # Get code size (address already on stack)
+            + Op.POP  # Discard code size result
+            # Increment counter
+            + Op.PUSH1[1]
+            + Op.ADD
+        ),
+        condition=Op.GT(num_contracts, Op.DUP1),  # num_contracts > counter
+    )
+    attack_code += Op.POP  # Clean up counter
+    # Store success flag for validation
+    attack_code += Op.SSTORE(SUCCESS_FLAG_SLOT, SUCCESS_FLAG_VALUE)
+
+    # Pre-initialize storage slot 0 to avoid cold SSTORE cost
+    attack_address = pre.deploy_contract(code=attack_code, storage={SUCCESS_FLAG_SLOT: 1})
+
+    # Attack transaction
+    attack_tx = Transaction(
+        to=attack_address,
+        gas_limit=gas_benchmark_value,
+        sender=pre.fund_eoa(),
+    )
+
+    # Post-state validation
+    post = {
+        # Verify attack completed successfully
+        attack_address: Account(storage={SUCCESS_FLAG_SLOT: SUCCESS_FLAG_VALUE}),
+    }
+
+    # Verify all pre-deployed contracts still exist with their code intact
+    # We check that they have nonce=1 (contracts were deployed)
+    for contract_address in deployed_contracts:
+        post[contract_address] = Account(
+            nonce=1,  # Contract exists and was deployed
+        )
+
+    # Calculate expected gas usage precisely
+    expected_benchmark_gas_used = (
+        intrinsic_gas
+        + memory_expansion_cost
+        + (num_contracts * cost_per_iteration)
+        + final_storage_cost
+    )
+
+    blockchain_test(
```

Comment:
I believe this test is similar to https://github.com/ethereum/execution-spec-tests/blob/562fde148440cf1a6a80305dd899dd15d91f4c1e/tests/benchmark/test_worst_stateful_opcodes.py#L122 (this test does similar things to BALANCE/EXTCODEHASH and also CALLs them)